### DISCUSSION TITLE
启动mysql插件失败

### DISCUSSION THREAD

**Discussion by Greenarrow1224** [emqx/emqx#6642] [Q&A] [answered]
错误信息：
{emqx_auth_mysql, {bad_return, {{emqx_auth_mysql_app,start,[normal,[]]}, {'EXIT', {{badmatch, {error, {shutdown, {failed_to_start_child,emqx_auth_mysql, {shutdown, {failed_to_start_child,worker_sup, {shutdown, {failed_to_start_child, {worker,1}, {{auth_method,<<"caching_sha2_password">>}, [{mysql_protocol,build_handshake_response,5, [{file,"mysql_protocol.erl"},{line,386}]}, {mysql_protocol,handshake,7, [{file,"mysql_protocol.erl"},{line,72}]}, {mysql_conn,init,1,[{file,"mysql_conn.erl"},{line,111}]}, {gen_server,init_it,2,[{file,"gen_server.erl"},{line,374}]}, {gen_server,init_it,6,[{file,"gen_server.erl"},{line,342}]}, {proc_lib,init_p_do_apply,3, [{file,"proc_lib.erl"},{line,249}]}]}}}}}}}}}, [{emqx_auth_mysql_app,start,2, [{file,"emqx_auth_mysql_app.erl"},{line,38}]}, {application_master,start_it_old,4, [{file,"application_master.erl"},{line,277}]}]}}}}}

**Comment by hjianbo** [comment, marked as answer]
Hi, @Greenarrow1224 Which emqx version was used? This error is a compatibility issue with MySQL 8.x.

You can check the FAQ docs to fix this issue first: https://docs.emqx.cn/enterprise/v4.4/faq/error.html#emq-x-无法连接-mysql-8-0
